parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh

```diff
@@ -25,9 +25,41 @@ installDeps() {
     fi
 }
 
+addMarinerNvidiaRepo() {
+    MARINER_NVIDIA_REPO_FILEPATH="/etc/yum.repos.d/mariner-nvidia.repo"
+    touch "${MARINER_NVIDIA_REPO_FILEPATH}"
+    cat << EOF > "${MARINER_NVIDIA_REPO_FILEPATH}"
+[mariner-official-nvidia]
```

Comment:
any reason not to add this in vhd?

Reply:
the cuda rpm is currently not supported in the Mariner core repository. Let me confirm with others in the team to make sure whether to pre-install this repo file in the Mariner image.

Reply:
for clarity I just mean in the AKS build VHD in install-dependencies.sh, not the base marketplace image

Reply:
Have moved this function to the VHD phase